src/Microsoft.AspNet.Mvc.Core/Formatters/OutputFormatter.cs

```diff
@@ -41,6 +43,32 @@ protected OutputFormatter()
         public MediaTypeCollection SupportedMediaTypes { get; }
 
         /// <summary>
+        /// Gets a cache with of the <see cref="SupportedMediaTypes"/> with utf-8 encoding
+        /// that this formatter uses when it writes the Content-Type header of the response.
+        /// </summary>
+        protected virtual IDictionary<string, string> OutputMediaTypeCache
```

Comment:
Id' like people that inherit from OutputFormatter to be able to reuse the cache if they need to, however, I've followed your comment and made it private.

In Web API there were a few components that maintained private caches and the fact that those caches were private made it impossible for them to properly derive from the class.

The reason I like it to be protected is because if it's important enough for us to have a cache for these types for performance reasons, it will be good for someone else to be aware of its existence, which is what protected does here.